examples/stateful/main.tf

```diff
@@ -216,3 +209,16 @@ resource "kubernetes_storage_class" "storage_class" {
     type      = "gp3"
   }
 }
+
+resource "kubernetes_storage_class_v1" "efs" {
```

Comment:
Just curious, what's the reason we need this now ?

Reply:
it looks like we didn't fill out the example completely/thoroughly (and there is still plenty of room for improvements) but we have the [EFS CSI addon](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/0848d87e4ab52f76dfff2e3d0c6d64d196b6e877/examples/stateful/main.tf#L73) enabled but we never demonstrated how it maps to the StorageClass like we do between the EBS CSI driver and its associated StorageClass resource. So this just demonstrates one way of consuming the EFS system using the dynamic provisioner method 

tl;dr - do we need it - the addon can be enabled without it but users wouldn't be able to utilize an EFS file share (sort of like creating a cluster without a data plane - you can, it just isn't very useful nor how its usually used)